src/commands/tag.js

```diff
@@ -0,0 +1,59 @@
+import { GitRefManager } from '../managers/GitRefManager'
+import { E, GitError } from '../models/GitError.js'
+import { FileSystem } from '../models/FileSystem.js'
+import { join } from '../utils/join.js'
+import { cores } from '../utils/plugins.js'
+
+/**
+ * Create a lightweight tag.
+ *
+ * @link https://isomorphic-git.github.io/docs/tag.html
+ */
+export async function tag ({
+  core = 'default',
+  dir,
+  gitdir = join(dir, '.git'),
+  fs: _fs = cores.get(core).get('fs'),
+  name,
+  value = undefined,
+  force = false
+}) {
+  try {
+    const fs = new FileSystem(_fs)
+    const ref = 'refs/tags/' + name
+
+    // Resolve passed value
+    value = await GitRefManager.resolve({
+      fs,
+      gitdir,
+      ref: value || 'HEAD'
+    })
+
+    try {
+      // Attempt to get old value
+      const oldValue = await GitRefManager.resolve({ fs, gitdir, ref })
+      // If found and not forced overwriting
+      if (!force) {
+        if (oldValue !== value) {
+          // It exists and differ from passed value
+          throw new GitError(E.RefExistsError, { noun: 'tag', ref: name })
+        } else {
+          // Nothing to do, as the old value and the new one is the same
+          return
```

Comment:
Sorry, probably this behaviour is unwanted (it won't throw if the value of the tag won't change). Actually I wrote this part without checking the behaviour of canonical `git tag` 🤣. I'll remove this and simply use `fs.exists` instead.